android/capacitor/src/main/java/com/getcapacitor/plugin/util/HttpRequestHandler.java

```diff
@@ -302,14 +302,18 @@ public static Object parseJSON(String input) throws JSONException {
             if ("null".equals(input.trim())) {
                 return JSONObject.NULL;
             } else if ("true".equals(input.trim())) {
-                return new JSONObject().put("flag", "true");
+                return true;
             } else if ("false".equals(input.trim())) {
-                return new JSONObject().put("flag", "false");
+                return false;
             } else if (input.trim().length() <= 0) {
                 return "";
             } else if (input.trim().matches("^\".*\"$")) {
                 // a string enclosed in " " is a json value, return the string without the quotes
                 return input.trim().substring(1, input.trim().length() - 1);
+            } else if (input.trim().matches("^-?\\d+(\\.\\d+)?$")) {
+                return Double.parseDouble(input.trim());
+            } else if (input.trim().matches("^-?\\d+$")) {
```

Comment:
integers also match the double regex check, so the integer else check should be moved up or they will be returned as double
or we can remove the integer check and only return doubles, I don't think that would be a problem on the JS side as there all the double and integers are "number"

Reply:
We will switch the order in case someone is using this class natively